tracing-attributes/src/attr.rs

```diff
@@ -333,7 +334,8 @@ impl ToTokens for Field {
             // `instrument` produce empty field values, so changing it now
             // is a breaking change. agh.
             let name = &self.name;
-            tokens.extend(quote!(#name = tracing::field::Empty))
+            let tracing = todo!();
+            tokens.extend(quote!(#name = #tracing::field::Empty))
```

Comment:
This here, in `<Field as ToTokens>::to_tokens` is the tricky bit, since I can't access `args` from here.

There are ~~two~~ three options for solving this that I see right now:

- I could turns this block into an inherent `impl Field`, add an `args: &InstrumentArgs` parameter and additionally add `fn to_token_stream(&self, args: &InstrumentArgs) -> TokenStream` for convenience.

- Alternatively, I could create wrappers for `Field` and `Fields` that additionally hold an `&'a InstrumentArgs`, which would be exported from this module instead of the parsed types.

- I could add `tracing` as private field to `Field` and set it at the end of `<InstrumentArgs as Parse>::parse`.

Personally, I would go with the first option since it seems cleanest, but it could lead to more awkward code where `Field`s are pasted since that implementation detail would be exposed outside `attr` and `Field` and `Fields` couldn't be used directly in `quote!` anymore.

Reply:
I went with the wrapper option (but with a generic borrowing wrapper), which turned out to be very clean to implement.